internal/meshdetect/detector.go

```diff
@@ -211,27 +219,140 @@ func isPodInitContainerInjected(meshKind MeshKind, pod *corev1.Pod) bool {
+			return nil, err
+		}
+
+		for i := range partialServiceList.Items {
+			serviceList = append(serviceList, &partialServiceList.Items[i])
+		}
+
+		continueToken = partialServiceList.GetContinue()
+
+		if partialServiceList.RemainingItemCount == nil || *partialServiceList.RemainingItemCount == 0 {
+			break
+		}
+	}
+
+	return serviceList, nil
+}
+
+// listAllEndpoints lists all endpoints in all namespaces.
+// returns map: namespaced name of endpoints -> endpoints resource
+// example: client.ObjectKey{Namespace: "default", Name: "service1"} ->
+// &corev1.Endpoints{
+//		ObjectMeta: metav1.ObjectMeta {
+//			Namespace: "default",
+//			Name: "service1", ...
+// 		},
+//		Subsets: ...,
+//		...
+// }
+func (d *Detector) listAllEndpoints(ctx context.Context, pageSize int) (
+	map[client.ObjectKey]*corev1.Endpoints, error) {
+	endpointsMap := map[client.ObjectKey]*corev1.Endpoints{}
+	continueToken := ""
+	for {
+		partialEndpointsList := &corev1.EndpointsList{}
+		err := d.Client.List(ctx, partialEndpointsList, client.Limit(pageSize), client.Continue(continueToken))
+		if err != nil {
+			return nil, err
+		}
+
+		for i := range partialEndpointsList.Items {
+			ep := &partialEndpointsList.Items[i]
+			key := client.ObjectKey{Namespace: ep.Namespace, Name: ep.Name}
+			endpointsMap[key] = ep
+		}
+
+		continueToken = partialEndpointsList.GetContinue()
+
+		if partialEndpointsList.RemainingItemCount == nil || *partialEndpointsList.RemainingItemCount == 0 {
+			break
+		}
+	}
+	return endpointsMap, nil
+}
+
+// listAllPods lists all pods in all namespaces.
+// returns map: namespaced name of pod -> pod resource
+// example: client.ObjectKey{Namespace: "default", Name: "pod1"} ->
+// &corev1.Pod{
+//		ObjectMeta: metav1.ObjectMeta {
+//			Namespace: "default",
+//			Name: "service1", ...
+// 		},
+//		Spec: ...,
+//		...
+// }
+func (d *Detector) listAllPods(ctx context.Context, pageSize int) (
+	map[client.ObjectKey]*corev1.Pod, error) {
+	podMap := map[client.ObjectKey]*corev1.Pod{}
+	continueToken := ""
+	for {
+		partialPodList := &corev1.PodList{}
+		err := d.Client.List(ctx, partialPodList, client.Limit(pageSize), client.Continue(continueToken))
+		if err != nil {
+			return nil, err
+		}
+
+		for i := range partialPodList.Items {
+			pod := &partialPodList.Items[i]
+			key := client.ObjectKey{Namespace: pod.Namespace, Name: pod.Name}
+			podMap[key] = pod
+		}
+
+		continueToken = partialPodList.GetContinue()
+		if partialPodList.RemainingItemCount == nil || *partialPodList.RemainingItemCount == 0 {
+			break
+		}
+	}
+	return podMap, nil
+}
+
 // DetectServiceDistribution detects how many services are running under each mesh.
 func (d *Detector) DetectServiceDistribution(ctx context.Context) (*ServiceDistributionResults, error) {
+
 	ret := &ServiceDistributionResults{
 		MeshDistribution: map[MeshKind]int{},
 	}
 
-	// list all services
-	serviceList := &corev1.ServiceList{}
-	err := d.Client.List(ctx, serviceList)
+	// list all services.
```

Comment:
Those comments are redundant given a good naming of a methods.

Reply:
changed the comments here to a description of purpose of the whole listing part.